packages/asset-uploader/src/patch.ts

```diff
@@ -63,6 +61,14 @@ export const patchAssets = async (
 
     const fileNames = new Set(files.map((file) => file.name));
 
+    // restore file when undo is triggered
+    await prisma.file.updateMany({
+      where: { name: { in: Array.from(fileNames) } },
+      data: {
+        isDeleted: false,
+      },
+    });
+
     await prisma.asset.createMany({
       data: addedAssets
```

Comment:
Not related to this PR, but we can fix here: 

`projectId: asset.projectId` below is a vulnerability. Should be either `projectId: projectId`, or we can filter out assets where `asset.projectId !== projectId`